Bottom: 0bc39c2212dd2dbee9f5ff7c2d5c38d49102352d
Top:    55576bceeaa70785f02f104788cb3a1e4b1857c1
Author: Pavel Roskin <proski@gnu.org>
Date:   2013-11-05 10:53:00 -0500

Separate binary and source trees

---

diff --git a/.farmrc b/.farmrc
index 56899a0..a1e5b7e 100644
--- a/.farmrc
+++ b/.farmrc
@@ -1,4 +1,5 @@
-farm=$HOME/src/farm
+farmsrc=$HOME/src/farm
+farmbin=$HOME/src/farm-bin
 tags="v2.6.14 v2.6.15 v2.6.16 v2.6.17 v2.6.18 v2.6.19 v2.6.20 v2.6.21
 v2.6.22 v2.6.23 v2.6.24 v2.6.25 v2.6.26 v2.6.27 v2.6.28 v2.6.29 v2.6.30
 v2.6.31 v2.6.32 v2.6.33 v2.6.34 v2.6.35 v2.6.36 v2.6.37 v2.6.38 v2.6.39
diff --git a/farm-checkout b/farm-checkout
index ca91b3e..353f760 100755
--- a/farm-checkout
+++ b/farm-checkout
@@ -4,9 +4,9 @@ set -e
 . $HOME/.farmrc
 
 cd $HOME/src/linux
-mkdir -p $farm
+mkdir -p $farmsrc
 for tag in $tags; do
-	srctree=$farm/`echo $tag | sed 's,/,-,'`
+	srctree=$farmsrc/`echo $tag | sed 's,/,-,'`
 	echo -n "Checking out $srctree "
 	if test -d $srctree; then
 		echo -e "\e[33mEXISTS\e[0m"
diff --git a/farm-clean b/farm-clean
index 18e7aab..bc2f1d3 100755
--- a/farm-clean
+++ b/farm-clean
@@ -3,7 +3,7 @@
 set -e
 . $HOME/.farmrc
 
-for tree in $farm/*; do
+for tree in $farmsrc/*; do
 	test -d $tree/Documentation || continue
 	cd $tree
 	echo "Cleaning $tree"
diff --git a/farm-compile b/farm-compile
index 4c791ce..72af274 100755
--- a/farm-compile
+++ b/farm-compile
@@ -13,7 +13,7 @@ if test -d utils; then
 	target=driver
 fi
 
-for bintree in $farm/*; do
+for bintree in $farmbin/*; do
 	arch=
 	if test -r $bintree/configured-i386; then
 		arch=i386
diff --git a/farm-config32 b/farm-config32
index 56ed509..06b08e2 100755
--- a/farm-config32
+++ b/farm-config32
@@ -6,9 +6,9 @@ set -e
 arch=i386
 ccflags=-m32
 
-for srctree in $farm/*; do
+for srctree in $farmsrc/*; do
 	test -d $srctree/Documentation || continue
-	bintree=$srctree-$arch
+	bintree=$farmbin/$(basename $srctree)-$arch
 	echo -n "Configuring $bintree "
 	if test -r $bintree/configured-$arch; then
 		echo -e "\e[33mEXISTS\e[0m"
@@ -16,7 +16,7 @@ for srctree in $farm/*; do
 	fi
 	rm -rf $bintree
 	rm -rf $bintree-bad
-	mkdir $bintree
+	mkdir -p $bintree
 	cd $srctree
 	$make HOSTCC="$ucc $ucflags" KBUILD_OUTPUT=$bintree \
 		CONFIG_WIRELESS_EXT=y CONFIG_NET_RADIO=y CONFIG_IPW2100=y \
diff --git a/farm-config64 b/farm-config64
index 4189d8a..91195ea 100755
--- a/farm-config64
+++ b/farm-config64
@@ -6,9 +6,9 @@ set -e
 arch=x86_64
 ccflags=-m64
 
-for srctree in $farm/*; do
+for srctree in $farmsrc/*; do
 	test -d $srctree/Documentation || continue
-	bintree=$srctree-$arch
+	bintree=$farmbin/$(basename $srctree)-$arch
 	echo -n "Configuring $bintree "
 	if test -r $bintree/configured-$arch; then
 		echo -e "\e[33mEXISTS\e[0m"
@@ -16,7 +16,7 @@ for srctree in $farm/*; do
 	fi
 	rm -rf $bintree
 	rm -rf $bintree-bad
-	mkdir $bintree
+	mkdir -p $bintree
 	cd $srctree
 	$make HOSTCC="$ucc $ucflags" KBUILD_OUTPUT=$bintree \
 		CONFIG_WIRELESS_EXT=y CONFIG_NET_RADIO=y CONFIG_IPW2100=y \
diff --git a/farm-hardlink b/farm-hardlink
index b7273e6..3e9747e 100755
--- a/farm-hardlink
+++ b/farm-hardlink
@@ -3,4 +3,4 @@
 set -e
 . $HOME/.farmrc
 
-hardlink -t $farm
+hardlink -t $farmsrc
